u8timeseries/models/statistics.py

```diff
@@ -0,0 +1,198 @@
+from ..timeseries import TimeSeries
+import numpy as np
+from statsmodels.tsa.stattools import acf
+from scipy.stats import norm
+from statsmodels.tsa.seasonal import seasonal_decompose
+import matplotlib.pyplot as plt
+from typing import Tuple
+import math
+
+
+def check_seasonality(ts: 'TimeSeries', m: int = None, max_lag: int = 24, alpha: float = 0.05):
+    """
+    Returns whether the TimeSeries [ts] is seasonal with period [m] or not.
+
+    If [m] is None, we work under the assumption that there is a unique seasonality period, which is inferred
+    from the Auto-correlation Function (ACF).
+
+    TODO : Generalize to multiple seasonality periods.
+
+    :param ts: The TimeSeries to check for seasonality.
+    :param m: The seasonality period to check.
+    :param max_lag: The maximal lag allowed in the ACF.
+    :param alpha: The desired confidence level (default 5%).
+    :return: A tuple (season, m), where s is a boolean indicating whether the TimeSeries has seasonality or not
+             and m is the seasonality period.
+    """
+    if m is not None and m < 0:
+        raise ValueError('m must be a positive integer.')
+
+    n_unique = np.unique(ts.values()).shape[0]
+
+    if n_unique == 1:  # Check for non-constant TimeSeries
+        return False, 0
+
+    r = acf(ts.values(), nlags=max_lag)  # In case user wants to check for seasonality higher than 24 steps.
+
+    gradient = np.gradient(r)
+    gradient_signs_changes = np.diff(np.sign(gradient))
+
+    if m is None:
+        # Tries to infer seasonality from AutoCorrelation Function if no value of m has been provided.
+        # We look for the first positive significant local maximum of the ACF by checking the sign changes
+        # in the gradient.
+
+        # Local maximum is indicated by signs_change == -2.
+        if len(np.nonzero((gradient_signs_changes == -2))[0]) == 0:
+            return False, 0
+
+        candidate = np.nonzero((gradient_signs_changes == -2))[0].tolist()
+
+    else:
+        # Check for local maximum when m is user defined.
+        if m - 1 not in np.nonzero((gradient_signs_changes == -2))[0]:
+            return False, m
+
+        candidate = [m - 1]
+
+    # Remove r[0], the auto-correlation at lag order 0, that introduces bias.
+    r = r[1:]
+
+    # The non-adjusted upper limit of the significance interval.
+    band_upper = r.mean() + norm.ppf(1 - alpha/2)*r.var()
+
+    # Significance test, stops at first admissible value.
+    is_significant = False
+    i = 0
+    c = 0
```

Comment:
To avoid managing `c` and `i` (which makes me sweat), you could have something like:
```
for i, candidate in enumerate(candidates):
    stat = _bartlett_formula(r, candidate, len(ts))
    if (r[candidate] > stat * band_upper):
        return True, candidate + 1
return False, 0
```
That's assuming `candidate` has been renamed to `candidates` (which would be a good idea IMO).